src/main/daml/Daml/Finance/Instrument/StructuredProduct/AutoCallable/Instrument.daml

```diff
@@ -0,0 +1,155 @@
+    description : Text
+      -- ^ A description of the instrument.
+    referenceAssetId : Text
+      -- ^ The reference asset ID. For example, in case of an AAPL underlying this should be a valid
+      --   reference to the AAPL fixings to be used for the payoff calculation.
+    initialFixing : Decimal
+      -- ^ The initial fixing of the underlying.
+    couponBarrier : Decimal
+      -- ^ The coupon barrier level.
+    callBarrier : Decimal
+      -- ^ The barrier used to automatically call the instrument.
+    finalBarrier : Decimal
+      -- ^ The barrier used to determine the final redemption amount.
+    couponRate : Decimal
+      -- ^ The fixed coupon rate, either per annum or per coupon period (depending on the
+      --   dayCountConvention below).
+    observationSchedule : PeriodicSchedule
+      -- ^ The schedule for the observation dates. These are used to observe the barrier, determine
+      --   whether the instrument is automatically called and to determine the final redemption
+      --   amount.
+    periodicSchedule : PeriodicSchedule
+      -- ^ The schedule for the periodic coupon payments.
+    holidayCalendarIds : [Text]
+      -- ^ The identifiers of the holiday calendars to be used for the coupon schedule.
+    calendarDataProvider : Party
+      -- ^ The reference data provider to use for the holiday calendar.
+    dayCountConvention : DayCountConventionEnum
+      -- ^ The day count convention used to calculate day count fractions. For example: Act360.
+    currency : InstrumentKey
+      -- ^ The currency of the product. For example, if the product pays in USD this should be a USD
+      --   cash instrument.
+    notional : Decimal
+      -- ^ The notional of the product. This is the face value corresponding to one unit of the
+      --   product. For example, if one product unit corresponds to 1000 USD, this should be 1000.0.
+    observers : PartiesMap
+      -- ^ The observers of the instrument.
+    lastEventTimestamp : Time
+      -- ^ (Market) time of the last recorded lifecycle event. If no event has occurred yet, the
+      --   time of creation should be used.
+    prevEvents : [EventData]
+      -- ^ A list of previous events that have been lifecycled on this instrument so far.
+  where
+    signatory depository, issuer
+    observer Disclosure.flattenObservers observers
+
+    interface instance Claim.I for Instrument where
+      view = Claim.View with acquisitionTime = dateToDateClockTime $ daysSinceEpochToDate 0
+      getClaims Claim.GetClaims{actor} = do
+        -- get the initial claims tree (as of the instrument's acquisition time)
+
+        let
+          getCalendars = getHolidayCalendars actor calendarDataProvider
+          floatingRate = None
+          capRate = None
+          floorRate = None
+          fixingBusinessCenters = case floatingRate of
+            None -> []
+            Some fr -> error "floating rate not yet supported"
```

Comment:
With `floatingRate` being hardcoded to `None` on line 92, can you elaborate for the reason to have `case floatingRate of` instead of setting it directly to `[]` ?

Reply:
Good point. I was thinking about whether to include support floating rates (and a test for it), but decided against it for now. I have simplified the code now.